### PR TITLE
LPS-167112 KBArticle RSS url does not change with parameters

### PR DESCRIPTION
**Pull request by marcogalluzzi**
Hi, this PR solves the problem described in issue. I've found two problems and I've made some refactorization based on similar classes: _WikiGroupServiceConfiguration_, _WikiPageService_ and _RSSStrutsAction_

- LPS-167112 FIX add support for "atom_1.0", "rss_1.0" or "rss_2.0" formats
- LPS-167112 Removed duplicated attribute "rssFormat", using server properties instead of hardcoded default values
- LPS-167112 Changed method signature in KBArticleService to separate rss format in type+version
- LPS-167112 auto baseline for module knowledge-base-api
- LPS-167112 Using type+version instead of single rss format string
- LPS-167112 Removed unnecessary variables and casting String to Integer
- LPS-167112 FIX using correct request parameters

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

**Comment by marcogalluzzi** [comment]
ci:test:sf

**Comment by marcogalluzzi** [comment]
ci:test:sf

**Comment by marcogalluzzi** [comment]
ci:test:sf

**Comment by marcogalluzzi** [comment]
ci:test:sf

**Comment by marcogalluzzi** [comment]
ci:test:relevant

**Comment by Hanlf** [comment]
ci:test:relevant

**Comment by Hanlf** [comment]
Hi @marcogalluzzi, the [service-builder-jdk8/0/0](https://test-1-8.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/124532//consoleText) failure should be related to this PR, could you please take a look? thanks!

**Comment by marcogalluzzi** [comment]
ci:test:stable

File: modules/apps/knowledge-base/knowledge-base-service/src/main/java/com/liferay/knowledge/base/service/impl/KBArticleServiceImpl.java [hunk 1]
```diff
@@ -312,8 +312,8 @@ else if (status == WorkflowConstants.STATUS_APPROVED) {
 
 	@Override
 	public String getGroupKBArticlesRSS(
-			int status, int rssDelta, String rssDisplayStyle, String rssFormat,
-			ThemeDisplay themeDisplay)
+			int status, int max, String type, double version,
```
Comment (AliciaGarciaGarcia) [inline modules/apps/knowledge-base/knowledge-base-service/src/main/java/com/liferay/knowledge/base/service/impl/KBArticleServiceImpl.java:315]: A `gw buildService` is needed

**Review by AliciaGarciaGarcia — CHANGES_REQUESTED** [review]

**Comment by marcogalluzzi** [comment]
ci:stop

File: modules/apps/knowledge-base/knowledge-base-service/src/main/java/com/liferay/knowledge/base/service/impl/KBArticleServiceImpl.java (hunk 1 above)
Comment (marcogalluzzi) [reply modules/apps/knowledge-base/knowledge-base-service/src/main/java/com/liferay/knowledge/base/service/impl/KBArticleServiceImpl.java:315]: Done, thanks for pointing it out

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Review by AliciaGarciaGarcia — DISMISSED** [review]
This is the expected behavior??

https://user-images.githubusercontent.com/47524751/201097171-b88285de-89de-45f4-a0c4-46820b48c3bd.mp4

**Comment by marcogalluzzi** [comment]
> This is the expected behavior??
> 
>  Compartir.pantalla.-.2022-11-10.13_55_52.mp4

Yes @AliciaGarciaGarcia , this behaviour haven't been changed. What has been fixed is the content of the showing XML file.

**Review by AliciaGarciaGarcia — APPROVED** [review]

**Comment by AliciaGarciaGarcia** [comment]
ci:forward

**Pull request closed without merging**